flowpipe/graph.py

```diff
@@ -108,62 +108,95 @@ def add_node(self, node):
             LogObserver.push_message(
                 "Node '{0}' is already part of this Graph".format(node.name))
 
-    def evaluate(self, threaded=False, submission_delay=0.1, raise_after=None):
-        """Evaluate all Nodes.
+    def evaluate(self, mode="linear", skip_clean=False,
+                 submission_delay=0.1, raise_after=None):
+        """Evaluate all Nodes in the graph.
+
+        Sorts the nodes in the graph into a resolution order and evaluates the
+        nodes. Nodes that do not depend on each other can be parallelized
```

Comment:
Suggestion: "Evaluation can naturally be parallelized by utilizing the dependencies between the nodes. Please see the "mode" keyword for the options."